packages/beacon-node/src/node/nodejs.ts

```diff
@@ -74,6 +75,12 @@ enum LoggerModule {
   sync = "sync",
 }
 
+/**
+ * Short delay before closing db to give async operations sufficient time to finish
+ * and prevent "Database is not open" errors when shutting down beacon node.
+ */
+const DELAY_BEFORE_CLOSING_DB_MS = 1000;
```

Comment:
500ms seems sufficient in most cases but might be good to have some safety margin